File: res/code/hello_beer/hello_beer.c
// hello beer: drink 0.5 liter in 4 rounds
#include <time.h>
#include <stdio.h>

const float LITER_PER_SIP = 37.0 / 1000.0;

float sips2liter( num_sips ){
    return (float)( num_sips ) * LITER_PER_SIP;
}

int main(int argc, char *argv[]){

    const int ROUNDS = 4;
    float beer_l = 0.5;
    int sips_per_round[ROUNDS] = {3, 7, 4, 2};

    printf("I drink %.2f l in %d rounds:", beer_l, rounds)
    for (int idx = 0; idx < ROUNDS; idx++){
        int sips = sips_per_round[idx];
        printf("I drink %d sips.", sips);
        sleep( sips );
        beer_l -= sips2liter( sips );

        if(beer_l <= 0.0){
            beer_l = 0.0;
            printf("My beer is empty!");
            break;
        } else {
            printf("I have %.3f l beer left.", beer_l);
        }
    }

}

File: res/code/hello_beer/hello_beer.rs
// hello beer: drink 0.5 l in 4 rounds

use std::{thread, time};

const LITER_PER_SIP: f32 = 37e-3;

fn sips2liter( num_sips: i32 ) -> f32 {
    return (num_sips as f32) * LITER_PER_SIP;
}

fn main(){

	let mut beer_l = 0.5;
	let rounds = 4;
	let sips_per_round = vec![3, 7, 4, 2];

	println!("I drink {} l in {} rounds:", beer_l, rounds);
	for sips in sips_per_round {
		println!("I drink {} sips.\n", sips);
		//thread::sleep( time::Duration::from_secs(sips) );
		beer_l = beer_l - sips2liter( sips as i32 );
		if beer_l <= 0.0 {
			println!("My beer is empty!");
			break
		} else {
			println!("I have {} l beer left.", beer_l);
		}
	}
}
